central/group/datastore/internal/store/store_test.go

```diff
@@ -509,3 +510,46 @@ func (s *GroupStoreTestSuite) TestGetFiltered() {
 	s.NoError(err)
 	s.ElementsMatch(expectedGroups, actualGroups)
 }
+
+func (s *GroupStoreTestSuite) TestDefaultGroup() {
+	defaultGroup := &storage.Group{
+		RoleName: "admin",
+		Props: &storage.GroupProperties{
+			AuthProviderId: "defaultGroup1",
+			Id:             "some-id",
+		},
+	}
+	initialGroup := &storage.Group{
+		RoleName: "Manager",
+		Props: &storage.GroupProperties{
+			AuthProviderId: "defaultGroup1",
+			Key:            "something",
+			Value:          "someone",
+			Id:             "some-id-3",
+		},
+	}
+
+	// 0. Setting up an existing group for the auth provider "defaultGroup1".
+	s.NoError(s.sto.Add(initialGroup))
+
+	// 1. Add the default group.
+	s.NoError(s.sto.Add(defaultGroup))
+
+	// 2. Adding the group a second time should not work and yield a "AlreadyExists" error.
+	defaultGroup.GetProps().Id = "some-id-2"
+	err := s.sto.Add(defaultGroup)
+	s.Error(err)
+	s.ErrorIs(err, errox.AlreadyExists)
+
+	// 3. Updating the initially existing group to make it a default group should fail.
+	//		Fetch the group by its properties.
```

Comment:
That's an interesting styling approach : ).

Reply:
Learnt it from the best 😜